javascript_nodejs/50.contoso-cafe-bot/dialogs/shared/prompts/getUserNamePrompt.js

```diff
@@ -0,0 +1,166 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+const { TextPrompt } = require('botbuilder-dialogs');
+const { MessageFactory } = require('botbuilder');
+const { LuisRecognizer } = require('botbuilder-ai');
+const { UserProfile } = require('../stateProperties');
+const { InterruptionDispatcher } = require('../../interruptionDispatcher');
+
+// LUIS service type entry for the get user profile LUIS model in the .bot file.
+const LUIS_CONFIGURATION = 'getUserProfile';
+
+// LUIS intent names from ./resources/getUserProfile.lu
+const WHY_DO_YOU_ASK_INTENT = 'Why_do_you_ask';
+const GET_USER_NAME_INTENT = 'Get_user_name';
+const NO_NAME_INTENT = 'No_Name';
+const NONE_INTENT = 'None';
+const CANCEL_INTENT = 'Cancel';
+
+// User name entity from ./resources/getUserProfile.lu
+const USER_NAME = 'userName_patternAny';
+const TURN_COUNTER_PROPERTY = 'turnCounterProperty';
+const HAVE_USER_PROFILE = true;
+const NO_USER_PROFILE = false;
+const CONFIRM_CANCEL_PROMPT = 'confirmCancelPrompt';
+
+// This is a custom TextPrompt that uses a LUIS model to handle turn.n conversations including interruptions.
+module.exports = class GetUserNamePrompt extends TextPrompt {
+    /**
+     * Constructor. 
+     * 
+     * @param {String} dialogId Dialog ID
+     * @param {Object} botConfig Bot configuration
+     * @param {Object} userProfileAccessor accessor for user profile property
+     * @param {Object} conversationState conversations state
+     */
+    constructor(dialogId, botConfig, userProfileAccessor, conversationState, onTurnAccessor) {
+        if (!dialogId) throw ('Missing parameter. Dialog ID is required.');
+        if (!botConfig) throw ('Missing parameter. Bot configuration is required.');
+        if (!userProfileAccessor) throw ('Missing parameter. User profile property accessor is required.');
+        if (!conversationState) throw ('Missing parameter. Conversation state is required.');
+        
+        super (dialogId, async (turnContext, step) => { 
```

Comment:
this needs to be updated to the newer validation function syntax that accepts a single promptContext object and returns a boolean